api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -482,6 +482,8 @@ public PartitionSpec build() {
   static void checkCompatibility(PartitionSpec spec, Schema schema) {
     for (PartitionField field : spec.fields) {
       Type sourceType = schema.findType(field.sourceId());
+      ValidationException.check(sourceType != null,
+          "Cannot find source column for partition field: %s", field);
```

Comment:
Looks unrelated to Apache release?

Reply:
Yeah, looks like this was included by accident. I'll open a separate PR for it.